modules/artifact/src/main/java/org/mule/module/artifact/classloader/ScalaClassValueReleaser.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.artifact.classloader;
+
+import org.mule.runtime.module.artifact.api.classloader.MuleDeployableArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.RegionClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.ResourceReleaser;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Addresses some known leaks in usages of Scala language from plugins/apps.
+ * <p>
+ * IMPORTANT: this class is on a different package than the rest of the classes in this module. The reason of that is that this
+ * class must be loaded by each artifact class loader that is being disposed. So, it cannot contain any of the prefixes that force
+ * a class to be loaded from the container.
+ *
+ * @since 4.3, 4.2.3
+ */
+public class ScalaClassValueReleaser implements ResourceReleaser {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ScalaClassValueReleaser.class);
+
+  private final Field classValueMapField;
+
+  public ScalaClassValueReleaser() {
+    Field classValueMapField = null;
+    try {
+      classValueMapField = Class.class.getDeclaredField("classValueMap");
+    } catch (Throwable t) {
+      LOGGER.warn("Unable to initialize ScalaClassValueReleaser", t);
+    }
+
+    this.classValueMapField = classValueMapField;
+  }
+
+  @Override
+  public void release() {
+    try {
+      WeakHashMap classValueMap = getClassValueMap();
```

Comment:
shouldn't we verify that that the scala releaser could be succesfully intiialized so that no NPE is raised during each redeployment?